server/unittools.cpp

```diff
@@ -1394,6 +1394,10 @@ void bounce_unit(struct unit *punit, bool verbose, bounce_reason reason,
 
       handle_unit_orders(pplayer, &packet);
 
+      if (!punit) {
```

Comment:
This will not work. The pointer is set at the beginning of the function and is not modified when killing the unit; only the associated memory is freed, resulting in a dangling pointer (which may or may not crash depending on [your compiler](https://feross.org/gcc-ownage/) and runtime).

The usual solution is to store the unit ID, then query the unit again after executing orders.